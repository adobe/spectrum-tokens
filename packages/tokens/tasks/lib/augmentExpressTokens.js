/*
Copyright 2023 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

const deprecationObj = {
  deprecated: true,
  deprecated_comment:
    "Express will merge with Spectrum with the release of Spectrum 2.",
};

const findExpressValue = (tokenObj) => {
  if (typeof tokenObj === "object" && tokenObj !== null) {
    if (Object.hasOwn(tokenObj, "value")) {
      return {
        ...tokenObj,
        ...deprecationObj,
      };
    } else {
      const result = {};
      Object.keys(tokenObj).forEach((tokenName) => {
        result[tokenName] = findExpressValue(tokenObj[tokenName]);
      });
      return result;
    }
  }
};

const augmentExpressTokens = (tokenObj) => {
  if (typeof tokenObj === "object" && tokenObj !== null) {
    const result = {};
    Object.keys(tokenObj).forEach((tokenName) => {
      if (tokenName === "express") {
        result[tokenName] = findExpressValue(tokenObj[tokenName]);
      } else {
        result[tokenName] = augmentExpressTokens(tokenObj[tokenName]);
      }
    });
    return result;
  } else {
    return tokenObj;
  }
};

export default augmentExpressTokens;
