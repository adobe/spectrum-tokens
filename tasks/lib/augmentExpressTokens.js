const deprecationObj = {
  deprecated: true,
  deprecated_comment:
    "Express will merge with Spectrum with the release of Spectrum 2.",
};

const findExpressValue = (tokenObj) => {
  if (typeof tokenObj === "object" && tokenObj !== null) {
    if (tokenObj.hasOwnProperty("value")) {
      return {
        ...tokenObj,
        ...deprecationObj,
      };
    } else {
      const result = {};
      Object.keys(tokenObj).forEach((tokenName) => {
        result[tokenName] = findExpressValue(tokenObj[tokenName]);
      });
      return result;
    }
  }
};

const augmentExpressTokens = (tokenObj) => {
  if (typeof tokenObj === "object" && tokenObj !== null) {
    const result = {};
    Object.keys(tokenObj).forEach((tokenName) => {
      if (tokenName === "express") {
        result[tokenName] = findExpressValue(tokenObj[tokenName]);
      } else {
        result[tokenName] = augmentExpressTokens(tokenObj[tokenName]);
      }
    });
    return result;
  } else {
    return tokenObj;
  }
};

export default augmentExpressTokens;
