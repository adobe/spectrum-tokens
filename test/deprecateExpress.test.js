import test from "ava";
import augmentExpressTokens from "../tasks/lib/augmentExpressTokens.js";

const fixture = {
  foo: {
    express: {
      value: "bar",
    },
  },
  "corner-radius-100": {
    sets: {
      spectrum: {
        sets: {
          desktop: {
            value: "4px",
          },
          mobile: {
            value: "5px",
          },
        },
      },
      express: {
        sets: {
          desktop: {
            value: "6px",
          },
          mobile: {
            value: "8px",
          },
        },
      },
    },
  },
};

test("Deprecate Express should add deprecation metadata to Express tokens", (t) => {
  t.snapshot(augmentExpressTokens(fixture));
});
