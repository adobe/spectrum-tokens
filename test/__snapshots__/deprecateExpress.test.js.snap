// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deprecate Express should add deprecation metadata to Express tokens 1`] = `
{
  "corner-radius-100": {
    "sets": {
      "express": {
        "sets": {
          "desktop": {
            "deprecated": true,
            "deprecated_comment": "Express will merge with Spectrum with the release of Spectrum 2.",
            "value": "6px",
          },
          "mobile": {
            "deprecated": true,
            "deprecated_comment": "Express will merge with Spectrum with the release of Spectrum 2.",
            "value": "8px",
          },
        },
      },
      "spectrum": {
        "sets": {
          "desktop": {
            "value": "4px",
          },
          "mobile": {
            "value": "5px",
          },
        },
      },
    },
  },
  "foo": {
    "express": {
      "deprecated": true,
      "deprecated_comment": "Express will merge with Spectrum with the release of Spectrum 2.",
      "value": "bar",
    },
  },
}
`;
